src/System.Linq/src/System/Linq/Enumerable.cs

```diff
@@ -2865,6 +2978,23 @@ public bool Contains(TKey key)
                 } while (g != _lastGrouping);
             }
         }
+        
+        public IGrouping<TKey, TElement>[] ToArray()
```

Comment:
(If we want to make it public, which seems reasonable, it would need to go through the new API process).